src/model_signing/_signing/sign_certificate.py

```diff
@@ -136,7 +136,13 @@ def __init__(
         self._log_fingerprints = log_fingerprints
 
         if not certificate_chain_paths:
-            certificate_chain_paths = [pathlib.Path(p) for p in certifi.where()]
+            os_certs = certifi.where()
+            if isinstance(os_certs, list):
+                certificate_chain_paths = [pathlib.Path(p) for p in os_certs]
+            elif isinstance(os_certs, str):
+                certificate_chain_paths = [pathlib.Path(os_certs)]
```

Comment:
> If certifi.where() returns a single string

FWIW, the library code / type hint says it will return a str.

> This is a type error bug, due to the fact that strings in Python are sometimes lists and sometimes not.

Not sure I follow.

Reply:
Oh, if it's always a string then this is simpler, let me fix.

For the string is sometimes list, sometimes not:

```python
>>> isinstance("as", list)
False
>>> isinstance("as", str)                                                                                                                           
True
>>> isinstance(["as"], list)                                                                                                                            
True
>>> [x for x in "as"]
['a', 's']
```

In Haskell default string is always a list of characters, type checking as a list of characters. Bytestrings are just a bytestring, you cannot access the characters directly, the `in` expression does not give them one by one. I think the same is in other languages too.

Reply:
https://github.com/certifi/python-certifi/blob/3e7076557d66b598fcca462e422dc988f09a6802/certifi/core.py#L21